sky/backends/cloud_vm_ray_backend.py

```diff
@@ -4398,6 +4401,42 @@ def post_teardown_cleanup(self,
                     else:
                         raise
 
+                # Clean up custom multi networks, e.g. the subnets, firewalls,
+                # and vpc created for GCP GPUDirect TCPX
+                try:
+                    cloud.check_features_are_supported(
+                        handle.launched_resources, {
+                            clouds.CloudImplementationFeatures.
+                            CUSTOM_MULTI_NETWORK
+                        })
+                    provision_lib.cleanup_custom_multi_network(
+                        repr(cloud), cluster_name_on_cloud, config['provider'],
+                        failover)
+                except exceptions.NotSupportedError:
+                    pass
+                except Exception as e:  # pylint: disable=broad-except
+                    resource_cleaned_up = False
+                    if purge:
+                        msg = common_utils.format_exception(e, use_bracket=True)
+                        logger.warning(
+                            f'Failed to cleanup custom multi network. Skipping '
+                            f'since purge is set. Details: {msg}')
+                    else:
+                        raise
+
+                if resource_cleaned_up:
+                    try:
+                        self.remove_cluster_config(handle)
```

Comment:
For the `remove_cluster_config` operation, before this PR, it seems that in any exception cases, including `exceptions.NotSupportedError` and `exceptions.PortDoesNotExistError`, this will not be executed, is this expected?
In this PR, it takes the `exceptions.NotSupportedError` and `exceptions.PortDoesNotExistError` cases as normal cases and will remove the cluster config here.
BTW, if we remove the cluster config here, `_detect_abnormal_non_terminated_nodes` in the following blocks will not be executed, is this expected?
Please help double confirm the behavior here @Michaelvll . Thanks!

Reply:
Hi @cg505, could you please help confirm if the code change here is reasonable? Thanks!

Reply:
I am not to worried about it, we already have some issues here - see #5011.

Reply:
OK, updated to follow the previous logic.